utilities/transactions/pessimistic_transaction.cc

```diff
@@ -264,7 +264,14 @@ Status PessimisticTransaction::Commit() {
           "Commit-time batch contains values that will not be committed.");
     } else {
       txn_state_.store(AWAITING_COMMIT);
+      if (log_number_ > 0) {
+        dbimpl_->logs_with_prep_tracker()->MarkLogAsHavingPrepSectionFlushed(
```

Comment:
For unprepared, We only track the oldest log_number_ for that transaction via `MarkLogAsHavingPrepSectionFlushed` and `MarkLogAsContainingPrepSection`, which should be enough to ensure that the oldest log + any log after does not get removed.

I think some tests in transaction_test exercise this, but I'll add an explicit one in write_unprepared_transaction_test as well. Currently, transaction_test isn't exercising the unprepared paths very much because max_write_batch isn't set, but that'll change soon after this diff.